src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolutionResult.cs

```diff
@@ -943,6 +944,15 @@ private static bool HadLambdaConversionError(DiagnosticBag diagnostics, Analyzed
                         UnwrapIfParamsArray(parameter));
                 }
             }
+            else if (!argument.HasExpressionType() && argument.Kind == BoundKind.DefaultPendingInference)
+            {
+                diagnostics.Add(
+                     ErrorCode.ERR_DefaultIsBadArg,
+                     sourceLocation,
```

Comment:
I think if you revert the changes in this method the resulting diagnostics are of the same quality as for other expressions. In other words, `default` should be handled the same way as `null` for the purposes of this method.